todo_app/app.py

```diff
@@ -81,15 +96,22 @@ def write_required(func):
         def wrapper_write_required(*args, **kwargs):
             allowed = False
             cur_user = flask_login.current_user
+            app.logger.debug('Login disabled?: %s', config.LOGIN_DISABLED)
+            app.logger.debug('Role is: %s', cur_user.role)
```

Comment:
If you look at the errors in your e2e test, it says (emphasis mine):
>  File "/app/***/app.py", line 100, in wrapper_write_required
**app.logger.debug('Role is: %s', cur_user.role)**
  File "/usr/local/lib/python3.9/site-packages/werkzeug/local.py", line 347, in __getattr__
    return getattr(self._get_current_object(), name)
**AttributeError: 'AnonymousUserMixin' object has no attribute 'role'**

So it looks like this is a genuine error, caused by the fact that this logging throws an error if the LOGIN_DISABLED flag is true, since the user doesn't have a role. We could fix this by e.g. only logging if it's not disabled, or even printing slightly different depending on the setting:
```python
debug_msg = 'Login disabled' if config.LOGIN_DISABLED else f'Login enabled, role is {cur_user.role}'
app.logger.debug(debug_msg)
```

Note that to test this I had to wipe my session cookie so that it didn't have my cached role from before I switched LOGIN_DISABLED on!